drivers/flash/Kconfig

```diff
@@ -162,4 +162,14 @@ config SOC_FLASH_MCUX_DEV_NAME
 	help
 	  Specify the device name for the flash driver.
 
+config SOC_FLASH_WRITE_BLOCK_SIZE
+	int "write block size (fixed)"
+	depends on FLASH && (SOC_FLASH_NRF5 || SOC_FLASH_STM32 ||\
```

Comment:
I'd make it only depends on FLASH - just add default 0 or sth in case no "compatible" driver is enabled